app/controllers/users_controller.rb

```diff
@@ -18,9 +21,33 @@ def create
     end
   end
 
+  def edit
+  end
+
+  def update
+    if @user.update_attributes(user_params)
+      flash[:success] = "Profile updated"
+      redirect_to @user
+    else
+      render 'edit'
+    end
+  end
+
   private
 
   def user_params
     params.require(:user).permit(:name, :email, :password, :password_confirmation)
   end
+
+  def signed_in_user
+    unless signed_in?
+      store_location
+      redirect_to signin_url, flash: { warning: "Please sign in." } unless signed_in?
```

Comment:
Prefer single-quoted strings when you don't need string interpolation or special symbols.